packages/ui-extensions/src/surfaces/admin/extension-points.ts

```diff
@@ -1,10 +1,39 @@
 import type {RenderExtension} from '../../extension';
 
-import type {AnyComponent} from './shared';
-import type {StandardApi} from './api';
+import type {AnyComponent, Components} from './shared';
+import type {StandardApi, CustomerSegmentationTemplateApi} from './api';
+import {AnyComponentBuilder} from '../../shared';
+
+type CustomerSegmentationTemplateComponent = AnyComponentBuilder<
+  Pick<Components, 'CustomerSegmentationTemplate'>
+>;
 
 export interface ExtensionPoints {
   Playground: RenderExtension<StandardApi<'Playground'>, AnyComponent>;
+  'Admin::Customers::SegmentationTemplatesFirstTimeBuyers::Show': RenderExtension<
```

Comment:
@lemonmade we just haven't thought about it. currently, each extension point maps to a category of templates. 

<img width="1661" alt="Screenshot 2023-02-23 at 4 30 54 PM" src="https://user-images.githubusercontent.com/3925905/221035375-223b858a-aff3-4fce-aade-43f4e7a3c64d.png">

Our extension could accept a `templateCategory` parameter as part of the API and return the right templates accordingly. This would also solve the problem of the "all" category. We'd have to run the extension for each category (instead of running extensions for each extension point).

How does that sound? Would you also use a union type instead of an enum to represent the categories?

Reply:
We (Segmentation app) have to handle all categories. Admin will loop over a list of categories and render the extension with the current category as `api`. If our extension doesn't handle a category, the page will be blank. We'll make sure to handle all categories. This is maybe where a non-rendering extension could have been matter because we could have returned a map of templates where the key is the category.

For other 1P (Shopify Email, etc.) and 3P apps, their templates will be grouped under an "Apps" entry point. We will provide `apps` as category but they can basically ignore it.

Reply:
I think it would be very messy to go down the route of passing them a category they might ignore. It would result in the wrong template being shown for the category or blank content. It actually seems a lot better to separate these to different extension points they can specifically opt in to use.

Reply:
The category is really just for us. Can we have different private and public APIs?
On the Admin side we were thinking of populating categories just for our app (app ID or handle lookup).
Any other extension should go to the Apps category and the provided `templateCategory` doesn't matter.